### PR TITLE
Bug 1111310 - Allow to configure mozregression with config file(s)

### PR DESCRIPTION
**Pull request by jpigree**
mozregression try to load a configuration file located on
"~/mozregression.cfg" and takes its values as default values
(overridable by command line passed args).

### CODE REVIEW

File: mozregression/main.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 import mozinfo
 import datetime
 import sys
+import os.path
```
Comment (parkouss) [inline mozregression/main.py]: I think just "import os" is more common here

File: mozregression/main.py [hunk 2]
```diff
@@ -26,6 +29,30 @@
...
+    """
+    defaults = {}
+
+    if os.path.isfile(conf_name):
+        try:
+            config = SafeConfigParser()
+            config.read([conf_name])
+            defaults = dict(config.items("Defaults"))
+            print("%s loaded" % conf_name)
+        except (SystemError, Error) as err:
```
Comment (parkouss) [inline mozregression/main.py]: Why SystemError ? I understand Error (ConfigParser.Error), but not SystemError here.

File: mozregression/main.py [hunk 3]
```diff
@@ -26,6 +29,30 @@
...
+            print("Error while parsing %s =>  no custom default values\n%s"
+                  % (conf_name, str(err)))
+    else:
+        print("%s not found => no custom default values" % conf_name)
```
Comment (parkouss) [inline mozregression/main.py]: I don't think this message is useful - don't print anything when no config is read.

File: mozregression/main.py [hunk 4]
```diff
@@ -26,6 +29,30 @@
...
+    defaults = {}
+
+    if os.path.isfile(conf_name):
+        try:
+            config = SafeConfigParser()
+            config.read([conf_name])
+            defaults = dict(config.items("Defaults"))
+            print("%s loaded" % conf_name)
+        except (SystemError, Error) as err:
+            print("Error while parsing %s =>  no custom default values\n%s"
```
Comment (parkouss) [inline mozregression/main.py]: I think it could be useful to exit here if the file is not well formed.

File: tests/unit/test_main.py [hunk 5]
```diff
@@ -45,6 +46,32 @@ def test_get_usage(self, stdout):
         self.assertEqual(exitcode, 0)
         self.assertIn('usage:', output)
 
+    def test_get_erronous_cfg_defaults(self):
+        temp_file = tempfile.NamedTemporaryFile()
```
Comment (parkouss) [inline tests/unit/test_main.py]: I suspect that the use of tempfile.NamedTemporaryFile() is the cquse of the error on windows, becquse the file is open in read and write mode. After writing, you may have to to a seek(0) to rewind the file.
I suggest that you use instead:
conf = tempfile.mkstemp()
self.addCleanup(os.unlink, conf) # ensure file will be delete
with open(conf, 'w') as f:
    f.write(...) # write the conf
# now, execute the test on conf

File: mozregression/main.py (hunk 2 above)
Comment (jpigree) [inline mozregression/main.py]: Okay. I tried to reproduce the case when the safeConfigParser throws a SystemError but I can't. I removed it.

**Comment by jpigree** [comment]
Yeah. At long last. I have vanquished those windows specificities. :)

File: tests/unit/test_main.py [hunk 6]
```diff
@@ -45,6 +47,46 @@ def test_get_usage(self, stdout):
...
+    def test_get_erronous_cfg_defaults(self):
+        conf = tempfile.mkstemp()
+        handle = conf[0]
+        filepath = conf[1]
+        self.addCleanup(os.unlink, filepath)
+
+        with os.fdopen(handle, 'w') as conf_file:
+            conf_file.write('aaaaaaaaaaa [Defaults]\n')
+
+        try:
```
Comment (parkouss) [inline tests/unit/test_main.py]: Could you rewrite this with self.assertRaise() instead ? I find it much more readable:
with self.assertRaises(SystemExit):
   main.get_defaults()

File: tests/unit/test_main.py [hunk 7]
```diff
@@ -45,6 +47,46 @@ def test_get_usage(self, stdout):
         self.assertEqual(exitcode, 0)
         self.assertIn('usage:', output)
 
+    def test_get_erronous_cfg_defaults(self):
+        conf = tempfile.mkstemp()
```
Comment (parkouss) [inline tests/unit/test_main.py]: could you rewrite this as handle, conf = tempfile.mkstemp() ?

File: tests/unit/test_main.py [hunk 8]
```diff
@@ -45,6 +47,46 @@ def test_get_usage(self, stdout):
...
+        valid_values = {'profile': 'PROFILE',
+                        'http-timeout': '10.2',
+                        'app': 'fennec',
+                        'persists': 'true',
+                        'http-cache-dir': '/home/jp/.mozregression',
+                        'repo': 'mozilla-aurora',
+                        'persist': '~/mozpersist',
+                        'bits': '64'}
+
+        conf = tempfile.mkstemp()
```
Comment (parkouss) [inline tests/unit/test_main.py]: same here: handle, conf = tempfile.mkstemp()

**Pull request merged**